### PR TITLE
fix(mobile_api): fix and change api to match the updated database

### PR DESCRIPTION
**Pull request by LaurentMag**
# Description
change and fix mobile backend to function with the new database version

# Linked Issues
What are the issues fixed by this pull request ?
#585

# Changes
What does it change ? nodeJS backend
Is is a breaking change ? YES
Is is a new feature ? yes
Is is a patch, fix, hotfix ?

# Screenshots
N/A

# Tests
How has it been tested ?

- [x] Manually
- [x] Automated tests
- [ ] QA
- [ ] Other

# Additional information
N/A

### CODE REVIEW

File: yaki_backend/src/__test__/__mocks__/team.ts [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
+    teamName: "Red team",
     teamCaptainId: 1,
-    teamName: "Red Team",
+    teamCustomerId: 1,
+    teamActifFlag: true,
   },
   {
     teamId: 2,
+    teamName: "Blue team",
     teamCaptainId: 2,
```
Comment (profotoce59) [inline yaki_backend/src/__test__/__mocks__/team.ts]: not suppose to become a list ? a team take a list of captain (or the conversion is done after ?)

File: yaki_backend/src/__test__/team-service.test.ts [hunk 2]
```diff
@@ -18,17 +18,21 @@ describe("teamService", () => {
...
       },
     ];
 
     it("should return a team", async () => {
       jest.spyOn(teamRepository, "getTeamByTeamMateId").mockResolvedValueOnce([
         {
           teamId: 2,
-          teamCaptainId: 2,
           teamName: "Equipe 2",
+          teamCaptainId: 2,
```
Comment (profotoce59) [inline yaki_backend/src/__test__/team-service.test.ts:33]: supposed to be a list of captain as it's a many to many relation or the translation is done after ?

File: yaki_backend/src/features/team/team.repository.ts [hunk 3]
```diff
@@ -2,8 +2,6 @@ import {Client, QueryResult} from "pg";
 import {TeamDtoIn} from "./team.dtoIn";
 
 export class TeamRepository {
-
-
   getTeamByCaptainId = async (captainId: number) => {
```
Comment (profotoce59) [inline yaki_backend/src/features/team/team.repository.ts:5]: getTeams instead of getTeam

File: yaki_backend/src/__test__/__mocks__/team.ts (hunk 1 above)
Comment (LaurentMag) [reply yaki_backend/src/__test__/__mocks__/team.ts]: deleted file, not used anymore

File: yaki_backend/src/__test__/team-service.test.ts (hunk 2 above)
Comment (LaurentMag) [reply yaki_backend/src/__test__/team-service.test.ts:33]: ticket creation this test wasn't review during this change, serveral others backend function will need to changed given the new workflow

File: yaki_backend/src/features/team/team.dtoIn.ts [hunk 4]
```diff
@@ -1,11 +1,15 @@
 export class TeamDtoIn {
   teamId: number;
-  teamCaptainId: number;
   teamName: string;
+  teamCaptainId: number;
+  teamCustomerId: number;
+  teamActifFlag: boolean;
 
-  constructor(teamId: number, teamCaptainId: number, teamName: string) {
+  constructor(teamId: number, teamCaptainId: number, teamName: string, teamCustomerId: number, teamActifFlag: boolean) {
```
[the remark this answers is not in the document]
Comment (LaurentMag) [reply yaki_backend/src/features/team/team.dtoIn.ts:8]: will be reviewed with ticket #644

File: yaki_backend/src/features/team/team.repository.ts (hunk 3 above)
Comment (LaurentMag) [reply yaki_backend/src/features/team/team.repository.ts:5]: will be reviewed with ticket #644

Comment (LaurentMag) [reply yaki_backend/src/features/team/team.repository.ts:5]: will be reviewed with ticket #644

**Pull request merged**